app/src/main/java/com/keylesspalace/tusky/util/RickRoll.kt

```diff
@@ -0,0 +1,20 @@
+package com.keylesspalace.tusky.util
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import com.keylesspalace.tusky.R
+
+object RickRoll {
+    fun shouldRickRoll(context: Context, domain: String) =
+            context.resources.getStringArray(R.array.rick_roll_domains).contains(domain)
```

Comment:
I think this needs to be extended to also take subdomains into account (e.g. `endsWith` instead of full match), and it should be case insensitive.

Reply:
Although it'd need to be more careful than literally using `endsWith`, no? You wouldn't want to match partial domain matches, only subdomains of the rickrolled domain. (That is, mastodon.gab.com should be rickrolled, notactuallygab.com shouldn't be rickrolled.)

Reply:
Yes you are right!